src/include/thread_group.h

```diff
@@ -0,0 +1,56 @@
+/*-
+ * Copyright (c) 2014-2016 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * WT_THREAD --
+ *	Encapsulation of a thread that belongs to a thread group.
+ */
+struct __wt_thread {
+	WT_SESSION_IMPL *session;
+	u_int id;
+	wt_thread_t tid;
+#define	WT_THREAD_RUN		0x01
+	uint32_t flags;
+
+	/* The runner function used by all threads. */
+	int (*run_func)(WT_SESSION_IMPL *session, WT_THREAD *context);
+};
+
+#define	WT_THREAD_CAN_WAIT		0x01
+#define	WT_THREAD_PANIC_FAIL		0x02
+
+/*
+ * WT_THREAD_GROUP --
+ *	Encapsulation of a group of utility threads.
+ */
+struct __wt_thread_group {
+	uint32_t	 alloc;    /* Size of allocated group */
+	uint32_t	 max;      /* Max threads in group */
+	uint32_t	 min;      /* Min threads in group */
+	uint32_t	 current_threads;/* Number of active threads */
```

Comment:
I added a name field/arg so that different thread groups can have identifying names.